app/router.js

```diff
@@ -25,17 +50,25 @@ const Router = Ember.Router.extend({
 
 Router.map(function() {
     this.route('page-not-found', {path: '/*bad_url'});
-    this.route('index', {path: 'preprints'});
-    this.route('page-not-found', {path: 'preprints/page-not-found'});
-    this.route('submit', {path: 'preprints/submit'});
-    this.route('discover', {path: 'preprints/discover'});
-    this.route('content', {path: '/:preprint_id' });
-    this.route('provider', {path: 'preprints/:slug'}, function() {
-        this.route('content', {path: '/:preprint_id'});
-        this.route('discover');
+
+    if (provider) {
```

Comment:
Does provider get recalculated properly when window.location changes? I would assume it does, since it works, but it seems weird that a const value changed would cause this Router.map to be rerun, assuming the preprint app is still a singleton across the different providers